testing/gtest/units/TestBuffer.cpp

```diff
@@ -0,0 +1,165 @@
+#include <gtest/gtest.h>
+#include "encoding/buffer.h"
+#include "encoding/graalencoding.h"
+
+const char *testString = "Test string";
+const int len = strlen(testString);
+
+TEST(Buffer, BufferDefaultConstructor) {
+	Buffer buffer;
+
+	EXPECT_EQ(buffer.length(), 0);
```

Comment:
I'd like to see named tests that describes what the test is testing, and structure should follow Arrange, Act, Assert. Makes it a whole lot easier to understand what the tests are supposed to do